### DISCUSSION TITLE
sa_column(ForeignKey) not working

### DISCUSSION THREAD

**Discussion by adpmhel24** [fastapi/sqlmodel#1480] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
branch_id: Optional[int] = Field(
        sa_column=Column("branch_id", Integer, ForeignKey("branch.id")),
    )
```

### Description

How to use sa_column with foreignkey?

### Operating System

macOS

### SQLModel Version

0.0.6

### Python Version

3.9.3

**Comment by JLHasson** [comment]
Your code looks correct except I don't believe you need the `"branch_id"` parameter in Column. Also ensure that the branch table has a parameter just named `id`?

In general it's hard to debug issues like this with such little info, you should provide more info about the class that contains this field as well as the class you are linking to, or a minimal example that reproduces the issue not using your code.

**Comment by adpmhel24** [comment]
> Your code looks correct except I don't believe you need the `"branch_id"` parameter in Column. Also ensure that the branch table has a parameter just named `id`?
> 
> In general it's hard to debug issues like this with such little info, you should provide more info about the class that contains this field as well as the class you are linking to, or a minimal example that reproduces the issue not using your code.

When I run alembic there's no foreignkey in branch_id but when i use foreign_key parameter of the field, it will add the fk_contraints

**Comment by adpmhel24** [comment]
> Your code looks correct except I don't believe you need the `"branch_id"` parameter in Column. Also ensure that the branch table has a parameter just named `id`?
> 
> In general it's hard to debug issues like this with such little info, you should provide more info about the class that contains this field as well as the class you are linking to, or a minimal example that reproduces the issue not using your code.
<img width="845" alt="Screen Shot 2022-05-04 at 1 52 55 PM" src="https://user-images.githubusercontent.com/40779451/166629187-b6b2dae8-998c-4529-8b97-a4a0e7d5fe57.png">

same to this screenshot it removes my foreignkey when I use sa_column=Column(ForeignKey).

But if i use foreign_key in Field class, fk was added. but when when I use Column it removes my foreignkey, I just to add onupdate="CASCADE" to my column.

**Comment by zhangbc97** [comment, marked as answer]
The branch_id Field should be set in the model which "table=True"

```
class Model(SQLModel, table=True):
    branch_id: Optional[int] = Field(
        sa_column=Column("branch_id", Integer, ForeignKey("branch.id")),
    )
```
